### PR TITLE
Add Azure Terraform documentation

### PR DESCRIPTION
**Pull request by gonzalogarciajaubert**
This PR will add Azure Terraform documentation for the changes in PR https://github.com/boostercloud/booster/pull/985

### CODE REVIEW

File: docs/chapters/05_going-deeper.md [hunk 1]
```diff
@@ -166,6 +166,92 @@ source <path-to-your-bash-file> && boost deploy -e production
 
 Now just let the magic happen, Booster will create everything for you and give you back your app ready to use URL. 🚀
 
+#### Azure synth command
+
+Azure provider implement the experimental **Booster** `synth` command. This command will generate [Terrafom](https://www.terraform.io/) templates from your code. It will also generate needed files to deploy your Booster application using [cdktf](https://learn.hashicorp.com/tutorials/terraform/cdktf).
```
Comment (alvaroloes) [inline docs/chapters/05_going-deeper.md:171]: Question: What would happen if I call `boost synth` on the AWS provider?

Comment (gonzalogarciajaubert) [reply docs/chapters/05_going-deeper.md:171]: You will get this error: `Error: Error: Attempted to perform the 'synth' operation with a provider that does not support this feature, please check your environment configuration.`

**Review by alvaroloes — APPROVED** [review]

File: docs/chapters/05_going-deeper.md [hunk 2]
```diff
@@ -166,6 +166,92 @@ source <path-to-your-bash-file> && boost deploy -e production
...
+
+Running `synth` command, for example `Booster synth -e production` will generate following files:
```
Comment (alvaroloes) [inline docs/chapters/05_going-deeper.md:173]: @gonzalogarciajaubert The command is `boost` 
```suggestion
Running `synth` command, for example `boost synth -e production` will generate following files:
```

**Review by alvaroloes — CHANGES_REQUESTED** [review]

**Review by alvaroloes — APPROVED** [review]

File: docs/chapters/05_going-deeper.md [hunk 3]
```diff
@@ -166,6 +166,92 @@ source <path-to-your-bash-file> && boost deploy -e production
...
+<application>
+├── cdktf.out
+│   └── stacks
+│       └── <application name><environment name>
+│           └── cdk.tf.json
+```
+
+Now deploy the template:
+
+1. Run (only the first time)
```
Comment (alvaroloes) [inline docs/chapters/05_going-deeper.md:228]: I am not following here.... It seems we are repeating the steps here again.

**Review by alvaroloes — CHANGES_REQUESTED** [review]

File: docs/chapters/05_going-deeper.md (hunk 3 above)
Comment (gonzalogarciajaubert) [reply docs/chapters/05_going-deeper.md:228]: The first time we are talking about deploying from the user project point of view. You have everything in the root folder so you can do it. The next time is talking about what files you need to copy to your CI/CD environment to deploy (without the user project)

Comment (alvaroloes) [reply docs/chapters/05_going-deeper.md:228]: ah!
ok thanks

**Review by alvaroloes — APPROVED** [review]

File: docs/chapters/05_going-deeper.md [hunk 4]
```diff
@@ -166,6 +166,92 @@ source <path-to-your-bash-file> && boost deploy -e production
...
+#### Azure synth command
+
+Azure provider implement the experimental **Booster** `synth` command. This command will generate [Terrafom](https://www.terraform.io/) templates from your code. It will also generate needed files to deploy your Booster application using [cdktf](https://learn.hashicorp.com/tutorials/terraform/cdktf).
+
+Running `synth` command, for example `boost synth -e production` will generate following files:
+
+* A file `cdktf.json`: A basic json file to deploy your application using `cdktf`
+* A folder `cdktf.out`: with the Terraform templates.
+
+Booster deploy command for Azure will deploy your application using the generated templates.  You don't need to run the `synth` command for deploy your application, the `deploy` command will generate the templates before deploy for you.
```
Comment (javiertoledo) [inline docs/chapters/05_going-deeper.md:178]: One question here, if you've generated the templates and you run `boost deploy` will Booster generate and overwrite the templates or will it use the existing ones?

**Review by javiertoledo — APPROVED** [review]
awesome!

File: docs/chapters/05_going-deeper.md (hunk 4 above)
Comment (gonzalogarciajaubert) [reply docs/chapters/05_going-deeper.md:178]: It will overwrite the templates and generate a new zip file with the functions

**Pull request merged**